### PR TITLE
[TFLite] Support Metal GPU Delegate for macOS

### PR DESCRIPTION
**Pull request by asus4**
This PR supports Metal GPU delegate on macOS. 

- Removed UIKit dependencies
- Added dylib build for macOS
- Added `extern` to  `TFLGpuDelegateCreate` and `TFLGpuDelegateDelete`
  - This option is required to call API from TFLite UnityPlugin. ref #34450 

In this PR, I don't add `MTLFeatureSet_macOS_GPUFamily1_v4` or lower support since I don't have these macs. I'm not sure if we should support these. 
https://developer.apple.com/documentation/metal/mtlfeatureset?language=objc

### CODE REVIEW

**Comment by jdduke** [comment]
Looks pretty reasonable to me, @impjdi any thoughts?

File: tensorflow/lite/delegates/gpu/BUILD [hunk 1]
```diff
@@ -159,6 +162,15 @@ ios_static_framework(
     deps = [":metal_delegate"],
 )
 
+macos_dylib(
```
Comment (jdduke) [inline tensorflow/lite/delegates/gpu/BUILD]: Can you add a comment here about how to build, if any special flags are required? And maybe include something like "Note: Support for MacOS is best-effort at the moment.".

**Review by jdduke — CHANGES_REQUESTED** [review]

**Comment by impjdi** [comment]
@jdduke 

Looks reasonable to me too.  If this works and doesn't break existing stuff, I welcome this change :)

File: tensorflow/lite/delegates/gpu/BUILD (hunk 1 above)
Comment (asus4) [reply tensorflow/lite/delegates/gpu/BUILD]: @jdduke ok, adding recommended build options in the comment

**Comment by asus4** [comment]
@jdduke Thanks for the review, added the comment in the BUILD file

**Review by jdduke — APPROVED** [review]

**Pull request merged**